src/main/webapp/app/localvc/commit-details-view/commit-details-view.component.ts

```diff
@@ -64,71 +56,50 @@ export class CommitDetailsViewComponent implements OnDestroy, OnInit {
-                this.repositoryType = 'USER';
-                this.participation = participation;
-                this.retrieveAndHandleCommits();
-            });
-        }
-    }
     /**
      * Retrieves the commits for the participation and sets the current and previous commit.
      * If there is no previous commit, the template commit is chosen.
      * Finally the diff report is fetched.
      * @private
      */
     private retrieveAndHandleCommits() {
-        let retrieveCommitHistory;
-
-        if (this.repositoryType === 'USER') {
-            retrieveCommitHistory = this.programmingExerciseParticipationService.retrieveCommitHistoryForParticipation(this.participationId);
-        } else {
-            retrieveCommitHistory = this.programmingExerciseParticipationService.retrieveCommitHistoryForTemplateSolutionOrTests(this.exerciseId, this.repositoryType);
+        let commitInfoSubscription;
+        if (this.repositoryType) {
+            commitInfoSubscription = this.programmingExerciseParticipationService.retrieveCommitHistoryForTemplateSolutionOrTests(this.exerciseId, this.repositoryType);
+        } else if (this.participationId) {
+            commitInfoSubscription = this.programmingExerciseParticipationService.retrieveCommitHistoryForParticipation(this.participationId);
+        }
+        if (!commitInfoSubscription) {
+            return;
         }
 
-        this.commitsInfoSubscription = retrieveCommitHistory
+        commitInfoSubscription
             .pipe(
-                tap((commits) => {
-                    this.commits = commits.sort((a, b) => (dayjs(b.timestamp!).isAfter(dayjs(a.timestamp!)) ? 1 : -1));
-                    for (let i = 0; i < this.commits.length; i++) {
-                        const commit = this.commits[i];
-                        if (commit.hash === this.commitHash) {
-                            this.currentCommit = commit;
-                            if (i < this.commits.length - 1) {
-                                this.previousCommit = this.commits[i + 1];
-                            } else {
-                                // choose template commit
-                                this.isTemplate = true;
-                                this.previousCommit = this.commits[this.commits.length - 1];
-                            }
-                            break;
-                        }
+                map((commits) => commits.sort((a, b) => (dayjs(b.timestamp).isAfter(dayjs(a.timestamp)) ? 1 : -1))),
+                tap((sortedCommits) => {
+                    this.commits = sortedCommits;
+                    const foundIndex = this.commits.findIndex((commit) => commit.hash === this.commitHash);
+                    if (foundIndex !== -1) {
+                        this.currentCommit = this.commits[foundIndex];
+                        this.previousCommit = foundIndex < this.commits.length - 1 ? this.commits[foundIndex + 1] : this.commits[this.commits.length - 1];
```

Comment:
```suggestion
                        this.previousCommit = foundIndex < this.commits.length - 1 ? this.commits[foundIndex + 1] : this.commits.last();
```

Reply:
I think as of now this wouldn't compile - last() returns `CommitInfo | undefined`, but the type of `previousCommit` is `CommitInfo`. I could theoretically write `last()!`